### PR TITLE
Support e2e tests for callback module SimApp

### PR DESCRIPTION
**Pull request by chatton**
## Description

closes: #4405

The task added in this workflow will build a docker image and push to the registry: `ghrc.io/cosmos/ibc-go-callbacks-simd:${GIT_TAG}`

We can then run the manual E2E test and pass this image/tag

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/callbacks/testing/simapp/simd/main.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package main
+
+import (
+	"os"
+
+	"github.com/cosmos/cosmos-sdk/server"
+	svrcmd "github.com/cosmos/cosmos-sdk/server/cmd"
+
+	"github.com/cosmos/ibc-go/modules/apps/callbacks/testing/simapp"
+	"github.com/cosmos/ibc-go/modules/apps/callbacks/testing/simapp/simd/cmd"
```
Comment (chatton) [inline modules/apps/callbacks/testing/simapp/simd/main.go:10]: duplicate file with updated imports

File: modules/apps/callbacks/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,28 @@
+FROM golang:1.20-alpine3.18 as builder
```
Comment (DimitrisJim) [inline modules/apps/callbacks/Dockerfile:1]: Leaving this comment as reminder in case https://github.com/cosmos/ibc-go/pull/4398 gets merged first

Comment (DimitrisJim) [reply modules/apps/callbacks/Dockerfile:1]:
```suggestion
FROM golang:1.21-alpine3.18 as builder
```

**Comment by DimitrisJim** [comment]
#4417 fixes callback failure

**Review by DimitrisJim — APPROVED** [review]
changes lgtm, I don't think we can trigger workflow w/o it being merged yea?

**Comment by chatton** [comment]
> changes lgtm, I don't think we can trigger workflow w/o it being merged yea?

We can't run new workflows via the UI if they're being added, but you can run ones that are being changed by specifying the branch, think it's possible to do it via `gh` cli though, let me try real quick!

File: modules/apps/callbacks/testing/simapp/simd/cmd/root.go [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+	"github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/x/crisis"
+	genutilcli "github.com/cosmos/cosmos-sdk/x/genutil/client/cli"
+
+	dbm "github.com/cometbft/cometbft-db"
+	cmtcfg "github.com/cometbft/cometbft/config"
+	"github.com/cometbft/cometbft/libs/log"
+
+	"github.com/cosmos/ibc-go/modules/apps/callbacks/testing/simapp"
+	"github.com/cosmos/ibc-go/modules/apps/callbacks/testing/simapp/params"
```
[the remark this answers is not in the document]
Comment (colin-axner) [reply modules/apps/callbacks/testing/simapp/simd/cmd/root.go:34]: Do you think we should add a README to the `testing` folder in callbacks explaining that this is a copy/paste of the ibc-go simapp with callbacks deps/logic injected? Just thinking maybe we should note this somewhere?

File: modules/apps/callbacks/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,28 @@
+FROM golang:1.21-alpine3.18 as builder
+
+RUN set -eux; apk add --no-cache git libusb-dev linux-headers gcc musl-dev make;
```
Comment (colin-axner) [inline modules/apps/callbacks/Dockerfile:3]: I guess make isn't necessary anymore? Happy to leave in

**Review by colin-axner — APPROVED** [review]
Thanks @chatton! :heart:

File: modules/apps/callbacks/Dockerfile (hunk 4 above)
Comment (DimitrisJim) [reply modules/apps/callbacks/Dockerfile:3]: yeah, good point. Guess it wouldn't hurt to leave it if theres a good change a Makefile might be added.

File: modules/apps/callbacks/testing/simapp/simd/cmd/root.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (chatton) [reply modules/apps/callbacks/testing/simapp/simd/cmd/root.go:34]: Yes I think there's no harm!

**Pull request merged**